tests/unit/htdocs/class/database/XoopsDatabaseTest.php

```
@@ -0,0 +1,308 @@
+    {
+        $this->assertFalse($this->db->exec('INSERT INTO test VALUES (1)'));
+    }
+
+    // ---------------------------------------------------------------
+    // normalizeLimitStart tests (protected, test via reflection)
+    // ---------------------------------------------------------------
+
+    #[Test]
+    public function normalizeLimitStartNullLimitReturnsNullNull(): void
+    {
+        $ref = new \ReflectionMethod($this->db, 'normalizeLimitStart');
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '146', column '20'). (undefined)

(MissingImport)